Refactor Install/Enable/Disable Scripts

gabrielhartmann opened the pull request:
There was a lot of code duplication around reading environment files /
config in the install.sh enable.sh and disable.sh scripts.  There was
also some inconsistency around where logging ended up.  Now all logging
is redirected to the same log file consistently.  There is still some
boilerplate and some more deduplication could be done, but this is just
a first attempt at cleaning up the code and making it more readable.  It
should have no functional effect.

ahmetalpbalkan commented on scripts/dockerlib.sh:
```diff
@@ -0,0 +1,94 @@
+}
+
+get_log_file() {
+    echo $(get_log_dir)/docker-handler.log
```
might be do these returns safer like `echo "$(get_log_dir)/docker-handler.log"` with quotes.

ahmetalpbalkan commented on scripts/enable.sh:
```diff
@@ -77,11 +51,11 @@ else
-    echo "composing:"
+    log "composing:"
     echo $compose_up | yaml_dump
     mkdir -p /home/$azureuser/compose
```
not sure if usernames can contain spaces but having quotes around this mkdir/pushd/cd could also help bugs that could raise from dirnames with spaces.

johngossman replied:
Agree. I also asked Jeff at the time to make pulling the compose exe optional/dependent on whether there was compose content on the theory that is less like the whole script will fail or be slow due to network weather.

gabrielhartmann replied:
Adding the quotes to the directory operations.  Will have to think about the best way to make a compose installation behavior change.  That would be the only functional change in this commit.

ahmetalpbalkan commented on scripts/enable.sh:
```diff
@@ -77,11 +51,11 @@ else
+    log "composing:"
     echo $compose_up | yaml_dump
     mkdir -p /home/$azureuser/compose
     pushd /home/$azureuser/compose
```
what do you think about creating a subshell instead of pushd/popd:

```
...
composepath="/home/$azureuser/compose"
(
    cd $composepath
    do compose stuff
)
continue doing other things
```

ahmetalpbalkan commented:
LGTM overall. I'll wait other folks to review as well.

johngossman commented on scripts/enable.sh:
```diff
@@ -77,11 +51,11 @@ else
 fi
 
-curl -L https://github.com/docker/compose/releases/download/1.1.0-rc2/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
+curl -L https://github.com/docker/fig/releases/download/1.1.0-rc2/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
```
Uh...this one is going the wrong way now. I assume this is just a merge issue.

gabrielhartmann replied:
You're talking about the change from "compose" to "fig" right?  I have no idea how that changed.  I'll change it back.

johngossman commented on scripts/dockerlib.sh:
```diff
@@ -0,0 +1,91 @@
+file_name=${0##*/}
+
+timestamp() {
+    date +"%Y-%m-%d %H:%M:%S"
```
This is fine, probably good. I'm just always a little leery of lots of one-line functions that aren't actually used in multiple places. The argument is that it makes the code more abstract and thus more readable. It also makes the code more abstract and *less* readable.

gabrielhartmann replied:
I'll remove the timestamp function.  The principle that trivially short, only called once, functions shouldn't exist sounds good to me.  Most of the functions below don't fall into that category as they're called multiple times.  Some I just moved to a common place because I thought it made sense, e.g. the *_dump set of functions should be together.  Others I kept for the sake of neatness.  For example get_script_dir is called 5 times, get_config_dir 2 times, but get_log_dir only 1 time.  They all do roughly the same thing so I thought it made sense to have all of them.  Even those can be refactored further since they have mostly copy pasted content, but refactoring can go too far.

Let me know if you want to undo some of these changes.  In any case I'll get rid of the timestamp function.

ahmetalpbalkan replied:
Those only called once would probably be a better fit for a variable. We have other sort of timestamps as well (see last line) but we can embed just this guy to "log" I suppose and make an inline call just fine.

gabrielhartmann commented:
In the latest update I changed as many functions in the dockerlib.sh file to variables as possible, and updated the function calls in other files to variable references.

ahmetalpbalkan commented:
@gabrielhartmann can't merge this now after #20 b/c of merge conflicts arised.  can you please

    git checkout dockerlib
    git fetch origin
    git rebase -i origin/master
    git status -s # see files with U
    # resolve merge conflicts
    # git add [those files]
    git rebase --continue
    git push -f yourorigin/dockerlib

wow that was long.

gabrielhartmann commented:
Updated with changes from #20.

ahmetalpbalkan commented:
will take a last look and merge.

ahmetalpbalkan commented:
LGTM
